src/jgikbase/test/idmapping/mongo_controller.py

```diff
@@ -56,6 +58,17 @@ def __init__(self, mongoexe: Path, root_temp_dir: Path, use_wired_tiger: bool=Fa
 
         self._proc = subprocess.Popen(command, stdout=self._outfile, stderr=subprocess.STDOUT)
         time.sleep(1)  # wait for server to start up
+        self.client = MongoClient('localhost', self.port)
+        # check that the server is up. See
+        # https://api.mongodb.com/python/3.7.0/api/pymongo/mongo_client.html
+        #    #pymongo.mongo_client.MongoClient
+        self.client.admin.command('ismaster')
+
+        # get some info about the db
+        self.db_version = self.client.server_info()['version']
+        self.index_version = 2 if (semver.compare(self.db_version, '3.4.0') >= 0) else 1
```

Comment:
So I'm inferring from this code that mongo has two different versions for indexing the database, v2 after 3.4.0 and v1 before. FYI this definitely took a bit for me to realize what is going on here, so might be worth commenting a description.

Reply:
done